### PR TITLE
Add neovim_sticky strategy, reusing already open terminal

### LINKED ISSUE #707: Reuse existing terminal for Neovim strategy
I would like to be able to keep reusing the same terminal using Neovim strategy, so that my splits don't move around.

It should be doable by opening terminal in interactive shell mode instead of command mode, so that it does not terminate after the command finished, then sending commands to it using `:put ='test'` trick.

One issue might be how to detect if the terminal is ready to run tests again, as running a command while tests are running will likely swallow the input, while running `^C` after tests are finished will terminate the shell.

### PR DESCRIPTION
**Pull request by dkaszews**
Closes #707 

Make sure these boxes are checked before submitting your pull request:

- [x] Add fixtures and spec when implementing or updating a test runner
- [x] Update the README accordingly
- [x] Update the Vim documentation in `doc/test.txt`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -167,6 +168,15 @@ disable this behavior with:
 let g:test#echo_command = 0
 ```
 
+With Neovim sticky strategy, if additional test run is requested before
```
Comment (codeinabox) [inline README.md:171]:
```suggestion
With the Neovim sticky strategy, if an additional test run is requested before
```

File: README.md [hunk 2]
```diff
@@ -167,6 +168,15 @@ disable this behavior with:
...
+the previous one has finished, it will either wait or fail to run at all.
+You can customize this behavior with following options:
```
Comment (codeinabox) [inline README.md:173]:
```suggestion
You can customize this behavior with the following options:
```

File: doc/test.txt [hunk 3]
```diff
@@ -625,6 +631,11 @@ disable this behavior with
 >
   let g:test#echo_command = 0
 <
+To have Neovim sticky strategy abort any currently running tests before
+scheduling new one, use
```
Comment (codeinabox) [inline doc/test.txt:635]:
```suggestion
scheduling a new one, use:
```

**Review by codeinabox — COMMENTED** [review]
Thank you! This looks great, just some minor suggestions for the documentation

**Comment by dkaszews** [comment]
@codeinabox Thank you for feedback, applied all suggestions

File: autoload/test/strategy.vim [hunk 4]
```diff
@@ -73,16 +73,48 @@ function! test#strategy#vimproc(cmd) abort
...
+    call s:neovim_new_term(&shell)
+    let b:[l:tag] = 1
+    let l:buffers = getbufinfo(bufnr())
+    call win_gotoid(l:current_window)
+  else
+    if !get(g:, 'test#preserve_screen', 1)
+      let l:cmd = [&shell == 'cmd.exe' ? 'cls': 'clear'] + l:cmd
+    endif
+    if get(g:, 'test#neovim_sticky#kill_previous', 0)
+      let l:cmd = [""] + l:cmd
```
Comment (dkaszews) [inline autoload/test/strategy.vim:107]: This is Ctrl+C character, github doesn't display it correctly.

**Review by codeinabox — APPROVED** [review]
Thanks again!

**Pull request merged**